loser/loser.go

```diff
@@ -2,9 +2,9 @@
 
 package loser
 
-import "golang.org/x/exp/constraints"
+import "cmp"
 
-func New[E constraints.Ordered](lists [][]E, maxVal E) *Tree[E] {
+func New[E cmp.Ordered](lists [][]E, maxVal E) *Tree[E] {
```

Comment:
The `constraints.Ordered` from `golang.org/x/exp/constraints` is also included into Go 1.21 as `cmp.Ordered`. See https://pkg.go.dev/cmp#Ordered